packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -993,6 +993,99 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ** See code in examples/api/lib/widgets/transitions/animated_builder.0.dart **
 /// {@end-tool}
 ///
+/// {@template flutter.flutter.animatedbuilder_changenotifier.rebuild}
+/// ## Granular rebuilds with AnimatedBuilder and ChangeNotifier
+///
+/// A [State.setState] method call requires Flutter to entirely rebuild the
+/// current widget plus a variable portion of the subtree, depending on the
+/// depth and the structure.
+///
+/// To avoid unnecessary work, the [AnimatedBuilder] class can be used along
+/// with a [ChangeNotifier] to only rebuild certain portions a widget,
```

Comment:
Any Listenable works. Maybe list the ChangeNotifier as a concrete example, but we should also say that any Listenable will work.